File: Tests/TestProjects/Shared/Operation Tests/AKOscillatorTests.m
//
//  AKOscillatorTests.m
//  iOSObjectiveCAudioKit
//
//  Created by Aurelius Prochazka on 5/22/15.
//  Copyright (c) 2015 Aurelius Prochazka. All rights reserved.
//

#import "AKTestCase.h"

#define testDuration 10.0

@interface TestOscillatorInstrument : AKInstrument
@end

@implementation TestOscillatorInstrument

- (instancetype)init
{
    self = [super init];
    if (self) {


        AKOscillator *frequencyOscillator = [AKOscillator oscillator];
        frequencyOscillator.frequency = akp(2);

        AKOscillator *oscillator = [AKOscillator oscillator];
        oscillator.frequency = [[frequencyOscillator scaledBy:akp(110)] plus:akp(440)];

        [self setAudioOutput:oscillator];
    }
    return self;
}

@end

@interface AKOscillatorTests : AKTestCase
@end

@implementation AKOscillatorTests

- (void)testOscillator
{
    // Set up performance
    TestOscillatorInstrument *testInstrument = [[TestOscillatorInstrument alloc] init];
    [AKOrchestra addInstrument:testInstrument];
    [testInstrument playForDuration:testDuration];

    // Check output
    XCTAssertEqualObjects([self md5ForOutputWithDuration:testDuration], @"95184436da24aa6c7b65b80b48f916e8");
}

@end

File: AudioKit/iOS/AudioKitTests/AKOscillatorTests.swift
//
//  AKOscillatorTests.swift
//  AudioKit
//
//  Created by Aurelius Prochazka on 8/4/16.
//  Copyright © 2016 AudioKit. All rights reserved.
//

import XCTest
@testable import AudioKit

class AKOscillatorTests: AKTestCase {

    var duration = 0.1
    
    func testOscillator() {
        let osc = AKOscillator(waveform: AKTable(.Sine, size: 4096))
        AudioKit.testOutput(osc, duration: duration)
        let expectedMD5 = "221e422c2ced547a391a18900ef08516"
        let md5 = AudioKit.tester!.MD5
        XCTAssertEqual(expectedMD5, md5)
    }

}
